### DISCUSSION TITLE
Case-Sensitive Binding of C# Properties or Methods for JavaScript Access in Puerts

### DISCUSSION THREAD

**Discussion by herki18** [Tencent/puerts#1905] [Q&A]
I am looking for a way to bind C# properties or methods to JavaScript in a case-sensitive manner using Puerts. Specifically, I would like to access properties in JavaScript with lowercase names even if they are defined with uppercase names in C#.  

For example:  

```csharp
[JSName("color")]
public Color Color { get; set; }
```

With this, on the JavaScript side, I want to access the property as `class.color` instead of `class.Color`.  

My main questions are:  
1. Is there a built-in way in Puerts to achieve this case-sensitive binding?  
2. If not, is there an attribute or configuration I can use to specify the JavaScript-facing name of a property or method?  
3. Are there any other workarounds or best practices in Puerts to accomplish this?  

Any guidance or suggestions would be greatly appreciated!

**Comment by chexiongsheng** [comment]
index.d.ts Generator:  https://github.com/Tencent/puerts/blob/master/unity/Assets/core/upm/Editor/Src/Generator/DTS.cs#L279, change the property / method name
`PuertsDLL.RegisterProperty`  register property to js engine and `PuertsDLL.RegisterFunction` register method to  js engine, change the name argument.
